Source/WebCore/page/Location.cpp

```diff
@@ -255,6 +255,10 @@ ExceptionOr<void> Location::replace(LocalDOMWindow& activeWindow, LocalDOMWindow
     if (!activeWindow.document()->canNavigate(dynamicDowncast<LocalFrame>(frame.get()), completedURL))
         return Exception { SecurityError };
 
+    // If the loader for the active window's browsing context (Frame) has no outgoing referrer, but has a parent
+    // navigable, then set the loader's outgoing referrer to the URL of the parent navigable's active document.
+    if (activeWindow.frame()->loader().outgoingReferrer().isEmpty() && frame->tree().parent())
+        activeWindow.frame()->loader().setOutgoingReferrer(dynamicDowncast<LocalFrame>(frame->tree().parent())->document()->urlForBindings());
```

Comment:
This seems wrong to me. What if you have three documents, a parent with two children, and one of the children navigates the other. Surely the referrer shouldn't be the parent?

https://html.spec.whatwg.org/#location-object-navigate looks at the incumbent, which I'm not sure is implemented in WebKit, but it's definitely not the parent. Not sure if there's something that approximates it.

I guess ideally in our code `setLocation()` would take care of it, similar to how it is in HTML. So all the individual callers don't have to try to account for it.